saleor/order/calculations.py

```diff
@@ -95,9 +95,6 @@ def fetch_order_prices_if_expired(
     # on the every recalculation
     handle_order_promotion(order, lines_info, database_connection_name)
 
-    # update `OrderLine.unit_discount_...` fields
-    update_unit_discount_data_on_order_line(lines_info)
-
```

Comment:
If we remove it from this function, I think we don't need to save `unitDiscount*` fields in save section (144-147). I think `base_unit_price_amount` is not updated here anymore as well

Reply:
Removed from save.